File: backend/youtube/responses/channelResponse.py
from typing import Optional
from pydantic import BaseModel


class ChannelResponse(BaseModel):
    provider: str
    publicId: str
    name: str
    internalImageUrl: Optional[str] = None
    subscriberCount: int = 0
    videoCount: int = 0
    viewCount: int = 0
    description: Optional[str] = None

File: frontend/dto/channelResponse.ts
import { z } from 'zod';

export const ChannelResponseSchema = z.object({
    provider: z.string(),
    publicId: z.string(),
    name: z.string(),
    internalImageUrl: z.string().nullable(),
    subscriberCount: z.number(),
    videoCount: z.number(),
    viewCount: z.number(),
    description: z.string().nullable(),
});

export type ChannelResponse = z.infer<typeof ChannelResponseSchema>;